pkgs/tools/misc/fzf/default.nix

```
@@ -16,8 +16,7 @@ let
   # so that using the shell completion (ctrl+r, etc) doesn't result in ugly
   # warnings on non-nixos machines
   ourPerl = if stdenv.isDarwin then perl else (
-    pkgs.writers.writeBashBin "perl" ''
-      #!${pkgs.runtimeShell}
+    writeShellScript "perl" ''
```

Sorry, I didn't actually test `writeShellScript` before suggesting it.
```suggestion
    writeShellScriptBin "perl" ''
```